docs/automated-testing/creating-custom-components.mdx

```diff
@@ -0,0 +1,192 @@
+---
+sidebar_label: Custom Bots and Discoverers
+---
+
+# Custom Bots and Discoverers
+
+If the built in Bot and Discoverer components aren't sufficient for your testing needs, you can also create your own with just a few steps and some code.
+
+## Building a Custom Discoverer
+
+Entity Discoverers are used to find entities in the scene that should be made available to automation.
+The built-in UI Element Discoverer is one example of an Entity Discoverer.
+It scans the scene for certain UI Elements (Buttons, Canvas Groups, etc.) and makes them available as entities.
+If your game has other entities you want to be able to automate, you can build an Entity Discoverer that will find them and provide information about their State and available Actions.
+
+### What is an Entity?
+
+An Entity represents a component in the scene that can be automated.
+Usually, an Entity maps to a single GameObject, but there's no requirement that there be an underlying GameObject for every entity.
+A single GameObject may be represented by multiple Entities, or an Entity may not be related to any GameObject at all.
+
+Entities have a very simple structure.
+They are represented by:
+
+* An ID that must be unique across all entity instances in the game
+* A name, usually derived from the GameObject name, that does not need to be unique.
+* A type, which is usually derived from the .NET Type Name of the Component backing the entity.
+* A description, which can be presented to users, **or** to Large Language Models (LLMs) to assist in understanding the purpose of the entity.
+* A set of States representing relevant properties of the entity
+* A set of Actions representing things the bot can _do_ with the entity.
+
+A State is a simple key/value pair that represents some value that will be relevant to a bot.
+For example, a Canvas Group exposes it's `interactable` property as a state value, to allow a bot to identify if the group is interactable.
+
+An Action is a function that can be called on the Entity itself.
+For example, a Button exposes a `click` action.
+When this action is called, a synthetic "click" event will be raised on the button and the appropriate handlers will run as if a user clicked the button.
+
+:::note
+Currently, Actions cannot take parameters. This will change in the future!
+:::
+
+### Creating a Discoverer
+
+To create a discoverer, create a new C# Script somewhere in your game (we recommend `Assets/Scripts/Automation`).
+Change the script class to inherit from `RegressionGames.Unity.Discovery.EntityDiscoverer`
+
+```csharp
+using RegressionGames.Unity.Discovery;
+
+namespace MyGame
+{
+    public class MyCustomDiscoverer : EntityDiscoverer
+    {
+    }
+}
+```
+
+An Entity Discoverer is a Unity component like any other, so if you want, you can provide handlers for any of Unity's events (`Awake`, `Update`, etc.).
+However, the base class provides a basic pattern we recommend following if you can.
```

Comment:
I like this inclusion of... here's our "official" pattern for how you should do it, but if that doesn't work for your game then you have access to these other standard Unity concepts. I think we should continue adding tidbits like this going forward!